### ISSUE TITLE
pkcs8: PrivateKeyInfo::from_der doesn't work with openssl-generated Ed25519 keys

### ISSUE THREAD

**Issue by akhilles** [RustCrypto/utils#133]
To reproduce, generate private key using:
```
openssl genpkey -algorithm ed25519 -outform der
```
and try to parse using `PrivateKeyInfo::from_der`

**Comment by tarcieri** [comment]
Interesting! I was just about to look into this.

**Comment by tarcieri** [comment]
This looks easy enough to fix. The ASN.1 structure looks like this:

```
    0:d=0  hl=2 l=  46 cons: SEQUENCE
    2:d=1  hl=2 l=   1 prim: INTEGER           :00
    5:d=1  hl=2 l=   5 cons: SEQUENCE
    7:d=2  hl=2 l=   3 prim: OBJECT            :ED25519
   12:d=1  hl=2 l=  34 prim: OCTET STRING      [HEX DUMP]:042017ED9C73E9DB649EC189A612831C5FC570238207C1AA9DFBD2C53E3FF5E5EA85
```

So where e.g. RSA keys use an explicit `NULL` for the `AlgorithmIdentifier` parameters, Ed25519 keys omit it entirely.

**Comment by akhilles** [comment]
ah! I thought it was some incompatibility between `OneAsymmetricKey` and `PrivateKeyInfo`.

**Comment by tarcieri** [comment]
Decoding is fixed in #134.

I'll submit a follow-up for fixing encoding after it's merged.

**Comment by tarcieri** [comment]
I fixed encoding in #135.

It involved special-casing the encoding of RSA keys which is a bit unfortunate, but there are now tests that ensure that the PKCS#8 encoder can reproduce the encoding of Ed25519 keys generated by OpenSSL.

**Comment by tarcieri** [comment]
Fixes released in `pkcs8` v0.2.2
